web2/hooks/useCheckIfOwner.ts

```diff
@@ -0,0 +1,23 @@
+import { INftItem } from "@/interfaces";
+import { multicall } from "@wagmi/core";
+import { useAuth } from "./useAuth";
+import { erc721ABI } from "wagmi";
+
+export const useCheckIfOwner = () => {
+  const { account } = useAuth();
+
+  return async (items: Pick<INftItem, "tokenAddress" | "tokenId">[]) => {
```

Comment:
I'd prefer to refactor this with [useContractReads](https://wagmi.sh/react/hooks/useContractReads) later on